src/Command/PurgeCommand.php

```diff
@@ -0,0 +1,318 @@
+            "SELECT id, title, leadPeriod, leadPurgeUploads FROM tl_form WHERE leadPeriod != ''"
+        );
+
+        foreach ($forms as $masterForm) {
+
+            $leadPeriodTime = $this->convertTimePeriodToTime($masterForm['leadPeriod']);
+            if (!empty($leads = $this->getAllLeads($masterForm['id'], $leadPeriodTime))) {
+                $deletedUploads = null;
+                if (!empty($leadsData = $this->getAllLeadsData($leads))) {
+                    $this->purgeLeadsData($leadsData, $masterForm);
+
+                    if ($masterForm['leadPurgeUploads']) {
+                        $deletedUploads = $this->purgeUploads($leadsData);
+                    }
+                }
+
+                $this->purgeLeads($leads, $masterForm);
+
+                // Add custom logic
+                if (isset($GLOBALS['TL_HOOKS']['postLeadsPurge']) && is_array($GLOBALS['TL_HOOKS']['postLeadsPurge'])) {
+                    foreach ($GLOBALS['TL_HOOKS']['postLeadsPurge'] as $callback) {
+                        if (is_array($callback)) {
+                            System::importStatic($callback[0])->{$callback[1]}(
+                                $masterForm,
+                                $leads,
+                                $leadsData,
+                                $deletedUploads
+                            );
+                        } elseif (is_callable($callback)) {
+                            $callback($masterForm, $leads, $leadsData, $deletedUploads);
+                        }
+                    }
+                }
+
+                $purged = true;
+            }
+        }
+
+        return $purged;
+    }
+
+    /**
+     * @param $timePeriod
+     * @return int
+     */
+    private function convertTimePeriodToTime($timePeriod)
+    {
+        $range = StringUtil::deserialize($timePeriod);
+
+        if (is_array($range) && isset($range['unit']) && isset($range['value'])) {
+            if (false !== ($timestamp = strtotime('- '.$range['value'].' '.$range['unit']))) {
+                return $timestamp;
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+     * @param $masterFormId
+     * @param $timestamp
+     * @return array
+     */
+    private function getAllLeads($masterFormId, $timestamp)
+    {
+        $leads = [];
+
+        $rows = $this->db->fetchAll(
+            "SELECT * FROM tl_lead WHERE master_id=? AND created<?",
+            [$masterFormId, $timestamp]
+        );
+
+        foreach ($rows as $row) {
+            $leads[$row['id']] = $row;
+        }
+
+        return $leads;
+    }
+
+    /**
+     * @param array $leads
+     * @param array $masterForm
+     * @return int
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function purgeLeads(array $leads, array $masterForm)
+    {
+        $deleted = 0;
+        $ids = implode(',', array_keys($leads));
+
+        if (!empty($ids)) {
+            $deleted = $this->db->executeUpdate(
```

Comment:
tried to use the `$this->db->delete(...)` but it's not compatible with `WHERE id IN(...)`